src/sp/transport/tcp/tcp.c

```diff
@@ -361,6 +361,11 @@ tcptran_pipe_recv_cb(void *arg)
 		goto recv_error;
 	}
 
+	if (p->closed) {
```

Comment:
Options here are to duplicate the relevant logic, or use a second label (and maybe reorder the statements at the label), or to add a check for NNG_ECLOSED -- but I think it might be worth doing that for the called function, because really we don't want to bump the error counter for NNG_ECLOSED from *any* caller.

Reply:
I'm thinking to merge this and add the change I want myself.